File: Day1/day1.js
const fs = require('fs')

fs.readFile('input.txt', (err, data) => {
	if (err) throw err;
	let nums = data.toString().split('\n');
	nums.map(parseInt);

	let counter = 1;
	for (let i = 0, n = nums.length - 2; i < n; i++) {
		if (nums[i + 1] > nums[i]) {
			counter++;
		}	
	}

	console.log(counter);
});

File: Day1/day1.py
def getNums():
    with open('input.txt') as file:
        tmp = file.read()

    return [int(x) for x in tmp.split('\n') if x != '']

def getCount(nums):
    counter = 0
    for i in range(len(nums) - 1):
        if nums[i + 1] > nums[i]:
            counter += 1

    print(counter)

def part1():
    nums = getNums()
    getCount(nums)

def part2():
    nums = getNums()
    windows = []
    for i in range(len(nums) - 2):
        win_sum = nums[i] + nums[i + 1] + nums[i + 2]
        windows.append(win_sum)

    getCount(windows)

if __name__ == "__main__":
    part1()
    part2()
